nml/version_info.py

```diff
@@ -121,13 +121,14 @@ def get_git_version(detailed = False):
         if tag:
             version = tag
             str_tag = tag
-        elif branch == "master":
-            version = isodate + "-g" + changeset
         else:
-            version = isodate + "-" + branch + "-g" + changeset
+            version = "0.5.0.dev" + isodate.replace("-", "") + "+"
```

Comment:
Is hardcoding the 'next' version here a good idea? I feel it's guaranteed that someone will forget to update it eventually.

Might be safer to have it use the most recent tag and `.post`.

Reply:
It's difficult to use the most recent tag. Traditionally NML created branches for the current stable version(s) starting with the major release of it and continued development on master. So the last tag might actually also mis-represent the actual version. Not sure there is a good solution to this problem.

 So changing the next major version to be released in master is not worst of ideas and only infrequently needs doing.

Reply:
And current master can be incompatible with most recent tag so it would be misleading.

Reply:
I assume what @FLHerne was refering to: find the first tag from HEAD, and add ".post0" behind that. This works for branches, and doesn't care that master can be incompatible.

I guess you can read "recent" in several ways. But in the way I explain, it should work just fine :)

Otherwise this piece of code is weird, and will result in annoying problems. Why else do this git-detection at all, and why not just hardcode the version in the first place? :D (either do it all one way, or all the other, I would say)